code/game/machinery/floodlight.dm

```diff
@@ -65,10 +65,21 @@
 		on = 1
 		set_light(brightness_on)
 
-	user.visible_message("<span class='notice'>[user] turns \the [src] [on ? "on":"off"]</span>", \
-	"<span class='notice'>You turn \the [src] [on ? "on":"off"]</span>")
+	user.visible_message("<span class='notice'>[user] turns \the [src] [on ? "on":"off"].</span>", \
+	"<span class='notice'>You turn \the [src] [on ? "on":"off"].</span>")
+
 	update_icon()
 
+/obj/machinery/floodlight/attack_ghost(var/mob/dead/observer/ghost)
+	if(blessed)
+		to_chat(ghost, "Your hand goes right through \the [src]...Is that some holy water dripping from it?")
+		return 0
+	if(!ghost.can_poltergeist())
```

Comment:
Checking if you can poltergeist also applies the cooldown if you can,
passing 0 for its arg prevents this behavior.